client/src/main/scala/drt/client/components/TerminalComponent.scala

```diff
@@ -55,128 +55,137 @@ object TerminalComponent {
-          )
-
-          val currentClass = if (props.terminalPageTab.mode == "current") "active" else ""
-          val snapshotDataClass = if (props.terminalPageTab.mode == "snapshot") "active" else ""
-          val planningClass = if (props.terminalPageTab.mode == "planning") "active" else ""
-          val staffingClass = if (props.terminalPageTab.mode == "staffing") "active" else ""
-
-          val currentContentClass = if (props.terminalPageTab.mode == "current") "fade in active" else "fade out"
-          val snapshotContentClass = if (props.terminalPageTab.mode == "snapshot") "fade in active" else "fade out"
-          val planningContentClass = if (props.terminalPageTab.mode == "planning") "fade in active" else "fade out"
-          val staffingContentClass = if (props.terminalPageTab.mode == "staffing") "fade in active" else "fade out"
-
-          val subMode = if (props.terminalPageTab.mode == "staffing") "desksAndQueues" else props.terminalPageTab.subMode
-
-          <.div(
-            <.ul(^.className := "nav nav-tabs",
-              <.li(^.className := currentClass,
-                <.a(^.id := "currentTab", VdomAttr("data-toggle") := "tab", "Current"), ^.onClick --> {
-                  GoogleEventTracker.sendEvent(props.terminalPageTab.terminal, "click", "Current")
-                  props.router.set(props.terminalPageTab.copy(
-                    mode = "current",
-                    subMode = subMode,
-                    queryParams = props.terminalPageTab.withUrlParameters(UrlDateParameter(None)).queryParams
-                  ))
-              }),
-              <.li(^.className := snapshotDataClass,
-                <.a(^.id := "snapshotTab", VdomAttr("data-toggle") := "tab", "Snapshot"), ^.onClick --> {
-                  GoogleEventTracker.sendEvent(props.terminalPageTab.terminal, "click", "Snapshot")
-                  props.router.set(props.terminalPageTab.copy(
-                    mode = "snapshot",
-                    subMode = subMode,
-                    queryParams = props.terminalPageTab.withUrlParameters(UrlDateParameter(None)).queryParams
-                  ))
-                }
-              ),
-              <.li(^.className := planningClass,
-                <.a(^.id := "planningTab", VdomAttr("data-toggle") := "tab", "Planning"), ^.onClick --> {
-                  GoogleEventTracker.sendEvent(props.terminalPageTab.terminal, "click", "Planning")
-                  props.router.set(props.terminalPageTab.copy(mode = "planning", subMode = subMode, queryParams = props.terminalPageTab.withUrlParameters(UrlDateParameter(None)).queryParams))
-                }
+
+      val staffAdjustmentRCP = SPACircuit.connect(_.maybeStaffDeploymentAdjustmentPopoverState)
+
+      <.div(
+        staffAdjustmentRCP(staffAdjustmentMP => {
+          staffAdjustmentMP() match {
+            case Some(dialogueState) => StaffAdjustmentDialogue(dialogueState)()
+            case _ => <.div()
```

Comment:
I think you can use VDomNode.empty() rather than making an empty div somewhere.

Reply:
VDomNode.empty() doesn't seem to be an option sadly. i'm almost certain there is a way to do it without an empty div but i've not been able to find it